### PR TITLE
Feature/view company profile screen

### PR DESCRIPTION
**Pull request by WalterGoldenOwl**
Description Pull Request:

## Todo
- Update UI and logic view company profile

## Task
Link of task: 

1. https://trello.com/c/OlbBzESs
2. https://trello.com/c/NRh1Cpn1

## Screenshot
Screenshot_1698216688
Screenshot_1698216693
Screenshot_1698216698
Screenshot_1698216701
Screenshot_1698216919

## Video

https://github.com/dustin-goldenowl/jobspot/assets/144195318/fdf1d344-a29f-4de6-82cc-805ae608ae9f

### CODE REVIEW

File: lib/src/core/config/router/app_router.dart [hunk 1]
```diff
@@ -169,5 +169,15 @@ class AppRouter extends $AppRouter {
           page: EditApplicantProfileRoute.page,
           transitionsBuilder: TransitionsBuilders.slideLeft,
         ),
+        CustomRoute(
+            path: '/viewCompanyProfile',
```
Comment (dustin-goldenowl) [inline lib/src/core/config/router/app_router.dart:173]: Can create a new branch for refactor this. Can put path into XPath

File: lib/src/presentations/home/screen/home_view.dart [hunk 2]
```diff
@@ -98,7 +98,8 @@ class HomeView extends StatelessWidget {
               onTap: () => HomeCoordinator.showViewJobDescription(job.id),
               onSave: () => context.read<HomeCubit>().saveJob(job.id),
               onApply: () => HomeCoordinator.showApplyJob(job: job),
-              isSave: PrefsUtils.getUserInfo()!.saveJob!.contains(job.id),
+              isSave:
+                  false, //PrefsUtils.getUserInfo()!.saveJob!.contains(job.id),
```
Comment (dustin-goldenowl) [inline lib/src/presentations/home/screen/home_view.dart]: Add comment or remove it `//PrefsUtils.getUserInfo()!.saveJob!.contains(job.id)`

File: lib/src/presentations/view_company_profile/cubit/view_company_profile_cubit.dart [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+  Future _getUserInfo(String uid) async {
+    final response = await _getUserInfoUseCase.call(params: uid);
+    if (response is DataSuccess) {
+      emit(state.copyWith(user: response.data));
+    }
+  }
+
+  Future favouritePost(FavouriteEntity entity) async {
+    final response = await _favouritePostUseCase.call(params: entity);
+    if (response is DataSuccess) {}
```
Comment (dustin-goldenowl) [inline lib/src/presentations/view_company_profile/cubit/view_company_profile_cubit.dart]: What's happen in this ?

File: lib/src/presentations/view_company_profile/cubit/view_company_profile_cubit.dart [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+    emit(state.copyWith(user: user.toUserEntity()));
+    final response = await _followUserUseCase.call(
+        params: FavouriteEntity(
+      uidTo: user.id,
+      id: state.user!.id,
+      listFavourite: user.follower,
+    ));
+    if (response is DataSuccess) {
+      // emit(state.)
```
Comment (dustin-goldenowl) [inline lib/src/presentations/view_company_profile/cubit/view_company_profile_cubit.dart]: Add comment

File: lib/src/presentations/view_company_profile/screen/view_company_profile_view.dart [hunk 5]
```diff
@@ -0,0 +1,295 @@
...
+          children: [
+            Container(height: 120, color: Colors.white),
+            Container(
+              height: 120,
+              width: double.infinity,
+              decoration: const BoxDecoration(color: Color(0xFFF3F2F2)),
+              child: Column(
+                children: [
+                  const SizedBox(height: 28),
+                  state.user != null
```
Comment (dustin-goldenowl) [inline lib/src/presentations/view_company_profile/screen/view_company_profile_view.dart]: Should be put it into Widget function. It's so clearly

File: lib/src/presentations/view_company_profile/screen/view_company_profile_view.dart [hunk 6]
```diff
@@ -0,0 +1,295 @@
...
+    BuildContext context, {
+    required List<String> follower,
+  }) {
+    bool isFollow = follower.contains(FirebaseAuth.instance.currentUser!.uid);
+    return CustomButtonProfile(
+      onTap: context.read<ViewCompanyProfileCubit>().followUser,
+      icon: isFollow
+          ? null
+          : Icon(FontAwesomeIcons.plus, color: AppColors.venetianRed, size: 15),
+      title: isFollow ? "Unfollow" : "Follow",
```
Comment (dustin-goldenowl) [inline lib/src/presentations/view_company_profile/screen/view_company_profile_view.dart]: Using .arb

File: lib/src/presentations/view_company_profile/screen/view_company_profile_view.dart [hunk 7]
```diff
@@ -0,0 +1,295 @@
...
+                borderRadius: BorderRadius.circular(10),
+              ),
+              controller: context.read<ViewCompanyProfileCubit>().tabController,
+              tabAlignment: TabAlignment.fill,
+              dividerColor: Colors.transparent,
+              indicatorSize: TabBarIndicatorSize.tab,
+              labelStyle: const TextStyle(fontWeight: FontWeight.w700),
+              onTap: onTap,
+              tabs: const [
+                Tab(text: 'About us'),
```
Comment (dustin-goldenowl) [inline lib/src/presentations/view_company_profile/screen/view_company_profile_view.dart]: The same above

File: lib/src/core/config/router/app_router.dart (hunk 1 above)
Comment (WalterGoldenOwl) [reply lib/src/core/config/router/app_router.dart:173]: I will fix it in feature

File: lib/src/presentations/view_company_profile/cubit/view_company_profile_cubit.dart (hunk 3 above)
Comment (WalterGoldenOwl) [reply lib/src/presentations/view_company_profile/cubit/view_company_profile_cubit.dart]: I forgot to do it 😅

File: lib/src/presentations/view_company_profile/cubit/view_company_profile_cubit.dart (hunk 4 above)
Comment (WalterGoldenOwl) [reply lib/src/presentations/view_company_profile/cubit/view_company_profile_cubit.dart]: I forgot delete it

File: lib/src/presentations/home/screen/home_view.dart (hunk 2 above)
Comment (WalterGoldenOwl) [reply lib/src/presentations/home/screen/home_view.dart]: I fixed it

File: lib/src/presentations/view_company_profile/screen/view_company_profile_view.dart (hunk 5 above)
Comment (WalterGoldenOwl) [reply lib/src/presentations/view_company_profile/screen/view_company_profile_view.dart]: I fix it

File: lib/src/presentations/view_company_profile/screen/view_company_profile_view.dart (hunk 6 above)
Comment (WalterGoldenOwl) [reply lib/src/presentations/view_company_profile/screen/view_company_profile_view.dart]: I fixed it

File: lib/src/presentations/view_company_profile/screen/view_company_profile_view.dart (hunk 7 above)
Comment (WalterGoldenOwl) [reply lib/src/presentations/view_company_profile/screen/view_company_profile_view.dart]: I fixed it

**Pull request merged**